pkg/monitortests/kubeapiserver/terminationlog/logs.go

```diff
@@ -0,0 +1,141 @@
+package terminationlog
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/openshift/origin/pkg/monitortestlibrary/nodeaccess"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type ClusterTerminationLogs struct {
+	errs []error
+}
+
+func NewClusterTerminationLogs(ctx context.Context, client kubernetes.Interface) *ClusterTerminationLogs {
+	return &ClusterTerminationLogs{}
+}
+
+func (p *ClusterTerminationLogs) Entries(ctx context.Context, client kubernetes.Interface) <-chan *LogEntry {
```

Comment:
I'm wondering if we could slightly simplify this function to something like:

```
func (p *ClusterTerminationLogs) Process(logHandler LogHandler) {
   masterNodes := getMasterNodes()
   for _, masterNode := range masterNodes {
     nodeTerminationLog := NewNodeTerminationLogs(masterNode)
     rawFile := nodeTerminationLog.Read()
     reader := bytes.NewReader(data) 
     scanner := bufio.NewScanner(reader) 
     for scanner.Scan() { 
        logHandler(scanner.Text()) 
     }
   }
}
```

WDYT ?

Reply:
We could even preprocess the line before calling the logHandler to extract some data like timestamp, for example:
```
for scanner.Scan() {
  line := scanner.Text()
  timestamp := extractTimestampFrom(line)
  structuredLogLine := {timestamp: timestamp, locator: nodeName/podName (?)
  logHandler(structuredLogLine)
}
```

Reply:
This method could actually accept an array of `logHandler LogHandler` so that we could register multiple logHandlers. Each handler could produce an interval. WDYT?